src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -99,7 +100,26 @@ public boolean hasPerson(Person person) {
 
     @Override
     public void deletePerson(Person target) {
+        requireNonNull(target);
+
+        if (addressBook == null) {
+            throw new AssertionError("Address book should not be null when deleting a person");
```

Comment:
It's not recommended to throw `AssertionError` for runtime control. 
```suggestion
            throw new IllegalStateException("Address book should not be null when deleting a person");
```
Or keep this as a true assert without relying on it in tests?